lib/app/service/ntp_service.dart

```diff
@@ -0,0 +1,107 @@
+import 'dart:async';
+import 'dart:ui';
+
+import 'package:app/app/service/service.dart';
+import 'package:injectable/injectable.dart';
+import 'package:logging/logging.dart';
+import 'package:ntp/ntp.dart';
+import 'package:rxdart/rxdart.dart';
+
+/// Maximum time to wait for NTP server response
+const _updateTimeout = Duration(seconds: 10);
+
+/// Period of time between NTP server requests
+const _updatePeriod = Duration(minutes: 1);
+
+/// Delay between app resume and NTP server request
+const _resumedDelay = Duration(milliseconds: 500);
+
+Timer? _updateTimer;
```

Comment:
I think, you can use `RefreshPollingQueue`+`RefreshingInterface` that provides same logic for periodic operation, see `currencies_service.dart`

Reply:
It's a good point but I'd prefer to keep the service clean of app and other packages dependencies because it's there is a secret plan (yeah, I want to refactor it!) to externalize the NtpService to separated package.